### DISCUSSION TITLE
PowerDNS-Admin / SAML / Okta setup example?

### DISCUSSION THREAD

**Discussion by jhaprins** [ngoduykhanh/PowerDNS-Admin#1296] [Q&A]
Hello,

I'm trying to setup a PowerDNS-Admin server with SAML authentication using Okta. Does anyone have examples on what to configure etc? At the moment I don't seem to get it working and the only thing I see is an error 500.
Besides that I see some evidence that there should be a SAML page in the configuration pages, but I can't find it, I only see things like Google, Microsoft, OpenID Connect. Do I need to do something special to enable the SAML configuration page?

Thanks,
Jan Hugo Prins

**Comment by jbe-dw** [comment]
Hello,

Unfortunately maintainers with knowledge in SAML have left, but maybe other users could help you with it.

Currently there is no interface for configuring SAML, it should be done with variables in config.py. There is a PR opened with it, that you could test #1189

**Comment by jhaprins** [comment]
That is really sad. 
I have tested the pull request and it doesn't apply cleanly. Have tried to just patch it, but get 2 reject files and 2 locations that tell me that it detect previously reversed patch.

patching file configs/development.py
Reversed (or previously applied) patch detected!  Assume -R? [n] y
Hunk #1 succeeded at 170 with fuzz 2 (offset 133 lines).
patching file 'docs/SAML Migration.md'
patching file docs/SAML.md
patching file powerdnsadmin/default_config.py
Hunk #1 succeeded at 29 (offset 2 lines).
patching file powerdnsadmin/models/setting.py
Reversed (or previously applied) patch detected!  Assume -R? [n] y
Hunk #2 FAILED at 110.
1 out of 2 hunks FAILED -- saving rejects to file powerdnsadmin/models/setting.py.rej
patching file powerdnsadmin/models/user.py
Hunk #1 succeeded at 649 (offset -3 lines).
Hunk #2 succeeded at 692 (offset -3 lines).
patching file powerdnsadmin/routes/admin.py
Hunk #1 succeeded at 1376 (offset 2 lines).
Hunk #2 succeeded at 1390 (offset 2 lines).
Hunk #3 succeeded at 1633 (offset 2 lines).
patching file powerdnsadmin/routes/index.py
Hunk #2 succeeded at 496 with fuzz 1 (offset 2 lines).
Hunk #3 succeeded at 592 (offset 2 lines).
Hunk #4 succeeded at 947 (offset 4 lines).
Hunk #5 succeeded at 998 with fuzz 1 (offset 5 lines).
Hunk #6 succeeded at 1018 (offset 5 lines).
Hunk #7 FAILED at 1026.
Hunk #8 succeeded at 1120 (offset 9 lines).
Hunk #9 succeeded at 1165 (offset 13 lines).
1 out of 9 hunks FAILED -- saving rejects to file powerdnsadmin/routes/index.py.rej
patching file powerdnsadmin/services/saml.py
Hunk #4 succeeded at 111 (offset 1 line).
Hunk #5 succeeded at 146 (offset 1 line).
Hunk #6 FAILED at 158.
1 out of 6 hunks FAILED -- saving rejects to file powerdnsadmin/services/saml.py.rej
patching file powerdnsadmin/templates/admin_setting_authentication.html
patching file powerdnsadmin/templates/errors/SAML.html
